runtime/src/builtins.rs

```diff
@@ -30,16 +32,34 @@ fn genesis_builtins() -> Vec<Builtin> {
     ]
 }
 
+#[derive(AbiExample, Debug, Clone)]
+pub enum ActivationType {
+    NewProgram,
+    NewVersion,
+}
+
 /// Builtin programs activated dynamically by feature
-fn feature_builtins() -> Vec<(Builtin, Pubkey)> {
-    vec![(
-        Builtin::new(
-            "secp256k1_program",
-            solana_sdk::secp256k1_program::id(),
-            solana_secp256k1_program::process_instruction,
+fn feature_builtins() -> Vec<(Builtin, Pubkey, ActivationType)> {
+    vec![
+        (
+            Builtin::new(
+                "secp256k1_program",
+                solana_sdk::secp256k1_program::id(),
+                solana_secp256k1_program::process_instruction,
+            ),
+            feature_set::secp256k1_program_enabled::id(),
+            ActivationType::NewProgram,
+        ),
+        (
+            Builtin::new(
+                "stake_program_v2",
```

Comment:
Partly due to https://github.com/solana-labs/solana/pull/13403#discussion_r518078470, basically you can't use same name for the updated new instruction_processor.

Snapshot restoration must discern whether it's new or old only from AccountDB's contents....